soc/arm/nuvoton_npcx/common/soc_rom_hdr.c

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2020 Nuvoton Technology Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/*
+ * Header for Nuvoton NPCX EC.
+ *
+ * This header is used by Nuvoton EC Booter (ROM Code).
+ */
+
+#include <stdint.h>
+#include <kernel.h>
+
+/* Signature used by fw header */
+#define SIG_FW_EC           0x2A3B4D5E
+
+/*
+ * Error detection addresses use by header
+ * (Set them zero means do error check automatically)
+ */
+#define ERRCHK_START_ADDR  0x0
+#define ERRCHK_END_ADDR    0x0
+
+/* FW Header used by NPCX5M5G Booter */
+struct __packed fw_header_t {
+	uint32_t anchor;         /* A constant used to verify FW header */
+	uint16_t ext_anchor;     /* Enable/disable firmware header CRC check */
+	uint8_t  spi_max_freq;   /* Spi maximum allowable clock frequency */
+	uint8_t  spi_read_mode;  /* Spi read mode used for firmware loading */
+	uint8_t  cfg_err_detect; /* FW load error detection configuration */
+	uint32_t fw_load_addr;   /* Firmware load start address */
+	uint32_t fw_entry;       /* Firmware entry point */
+	uint32_t err_detect_start_addr; /* FW error detect start address */
+	uint32_t err_detect_end_addr;   /* FW error detect end address */
+	uint32_t fw_length;      /* Firmware length in bytes */
+	uint8_t  flash_size;     /* Indicate SPI flash size */
+	uint8_t  reserved[26];   /* Reserved bytes */
+	uint32_t sig_header;     /* The CRC signature of the firmware header */
+	uint32_t sig_fw_image;   /* The CRC or Checksum of the firmware image */
+} __aligned(1);
```

Comment:
Can this __aligned() go next to the __packed?

Also why does it need to be byte-aligned? This increases code size, I suspect

But now I see that the header itself has no alignment - e.g. the start_addr field. So I guess this is just a design decision.

Reply:
Oops. It should be __packed. I also added a BUILD_ASSERT macro to check the header size is 64 bytes.